sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -386,6 +386,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DEFAULT_PARALLELISM = buildConf("spark.sql.default.parallelism")
+    .doc("The number of parallelism for Spark SQL, the default value is " +
+      "`spark.default.parallelism`.")
+    .version("3.1.0")
+    .intConf
+    .checkValue(_ > 0, "The value of spark.sql.default.parallelism must be positive.")
+    .createWithDefault(1)
```

Comment:
shall we make this optional? so that we can fallback to `spark.default.parallelism` at caller side.

Reply:
`.fallbackConf(...)` or `.createOptional`? +1 for @cloud-fan 's suggestion.